### PR TITLE
Combine Wind & Solar Target Generation Charts to One

### PR DESCRIPTION
**Pull request by swatland610**
Hey @nofurtherinformation ! 

Just created a quick PR related to your ask in slack! Should be a quick one, but let me know if you have any issues / concerns!

### CODE REVIEW

**Comment by vkoves** [comment]
@swatland610 - there's some linting failures, and also should we show a number for the amount of renewables already present above the graph, like we do for EVS?

| Power | Cars |
| --- | --- |
| Screenshot from 2022-06-04 13-19-17 | Screenshot from 2022-06-04 13-19-30 |

Lastly if your other PR is now no longer relevant, can you close it?

**Comment by SuragNuthulapaty** [comment]
When enough wind turbines or solar is made, it makes the wattage needed is negative instead of zero

172520965-d2bdd4cc-626e-46a3-99f8-09ab625f8e14

https://deploy-preview-111--state-decarbonize.netlify.app/iowa

**Comment by swatland610** [comment]
hey @vkoves ! sorry about that! I'll fix the linting, but as for your main question, I opted to keep things grounded in **installed capacity** or (**MWs**) as that's how most folks doing in depth research on this communicate it. 

Also, I feel really good about the math we did to get the current numbers. But if we try adding in figures of _how many wind turbines_ & _how many solar panels_, the math gets incredibly complex because we'd need to consider where all of these wind turbines and panels need to go and (where they actually can go). 

Given that our end goal visualization is just a bar chart, I think sticking with **MWs** is most appropriate. 

Let me know if that's not quite what you meant!

**Comment by swatland610** [comment]
@SuragNuthulapaty good catch! I noticed that two and this current PR of combining these two graphs _should_ fix those issues. 

Also, I adjusted the raw data so values shouldn't be going below zero. So once this gets merge, we should be all good!

**Comment by vkoves** [comment]
@swatland610 - I do actually still see the negative value issue on Idaho [using the latest Netlify](https://deploy-preview-111--state-decarbonize.netlify.app/iowa) so I think something is still off there:

Screenshot_20220608-081622

Maybe Idaho in particular has something weird

**Comment by vkoves** [comment]
And as for current values, I meant more so if we could show the value underlying the graph, as in a paragraph after the amount we need to build saying:

"State already has _ MWs of solar and _ MWs of wind, _% of what we need".

This is directly modeled off of the vehicles section, since I think writing out the value is helpful to get a sense of how far a state has moved in a more direct way.

**Comment by SuragNuthulapaty** [comment]
@vkoves you do mean Iowa right? I just checked and Idaho looks good

image

**Comment by vkoves** [comment]
Yes, sorry, I screenshoted and linked the right thing but commented the wrong one 😂

**Comment by swatland610** [comment]
Thanks y'all! I promise I'm on it. Just traveling back to the states over the next few days. I'll tackle this early next week!

TLDR; there's a few states like Iowa that are already in good shape as it comes to wind. 

So I think your suggestion @vkoves of altering the copy for those is a good one! I can take a stab and working that out and share when I'm done!

**Comment by vkoves** [comment]
@swatland610 - I've fixed up some stuff, including a clamp for the amount of solar and wind to build (so if it's supposedly negative we lock it at zero, since that just indicates we've built plenty) and also added some text indicating the current solar and wind capacity.

I also figured out the root cause of why the Idaho graph was full when it shouldn't have been (which I don't think I realized before) - we were adding two percentages together and subtracting them from 100, but that means that hitting both goals would gets us to 200%. To be more specific, we were doing `totalRemaining = 100 - (percSolarTarget + percWindTarget)`, which at 100% solar and 100% wind yields 200%, or in Idaho's case, 100% wind makes the total 100%.

I now think everything is good but I think one point of confusion is that since we're mapping these two renewable goals, states with similar percentages of fossil fuel emissions show different graphs. Illinois, for comparison, only gets 8.6% of it's energy from renewables, but we get 60% of our energy from nuclear (total 68.6% non-fossil power), so I would say we are a little further ahead at Idaho at 58% renewables and 6% nuclear (total 63.7% non-fossil power). Instead though the graph indicates that Idaho is way ahead of Illinois, since it has much more solar and wind specifically: 

| Iowa | Illinois |
| --- | --- |
| Screenshot from 2022-06-10 18-45-08 | Screenshot from 2022-06-10 18-45-01 |
| Screenshot from 2022-06-10 18-50-09 | Screenshot from 2022-06-10 18-50-26 |

@nofurtherinformation and @derekeder - I'd love your thoughts on this too. @nofurtherinformation I think your graph labelled this as a fossil vs non fossil graph but I'm not sure if you had updated the graph to pull different data to make that happen.

**Comment by vkoves** [comment]
Actually, reading more I think PR #99 will still be useful and needed, since it indeed switches us from just solar/wind to fossils non-fossil, it'll just have to be re-worked a bit once this PR is merged. So I think this PR is ready to go, but someone else should review it since I made some tweaks

**Comment by vkoves** [comment]
Note from Juan (perhaps for the future): Nobody knows what a MW is, so we should probably specify number of turbines and number of solar panels/sq. feet/acres of solar

**Comment by derekeder** [comment]
@vkoves to recap, you'd like to review and merge in this PR and then reopen PR #99 after, is that correct? any way we could bring in what we want from #99 here?

**Comment by vkoves** [comment]
@derekeder - based on our discussion I think this PR _alone_ is fine, since Juan was basically saying we care about building the solar and wind we need, not the overall green-ness of the state

**Review by derekeder — APPROVED** [review]
@vkoves cool. code looks good. i think we can merge this in
